include/range/v3/numeric/partial_sum.hpp

```diff
@@ -30,16 +30,14 @@ namespace ranges
     inline namespace v3
     {
         template<typename I, typename O, typename BOp = plus, typename P = ident,
-            typename V = value_type_t<I>,
-            typename X = concepts::Invocable::result_t<P&, V>,
-            typename Y = concepts::Invocable::result_t<BOp&, X, X>>
+            typename X = projected<projected<I, coerce<value_type_t<I>>>, P>,
+            typename Y = reference_t<X>>
         using PartialSummable = meta::strict_and<
+            SemiRegular<detail::decay_t<Y>>,
```

Comment:
> Perhaps it makes no sense for range library to try and prevent all the potential abuses?

That isn't an argument against a range library trying to prevent _some_ abuses. @CaseyCarter, how strenuously are you objecting to `Semiregular` here? I like strong constraints.